tests/plugins/remove_duplicates.test.ts

```diff
@@ -0,0 +1,190 @@
+    expect(getEvaluatedCell(model, "C3").value).toBe("C4");
+    expect(getEvaluatedCell(model, "C4").value).toBe("C6");
+  });
+
+  test("when formula take into account the evaluated cell value", () => {
+    setSelection(model, ["A2:C7"]);
+    model.dispatch("REMOVE_DUPLICATES", { columnsToAnalyze: [0], hasHeader: false });
+
+    expect(getEvaluatedCell(model, "C2").value).toBe("C2");
+    expect(getEvaluatedCell(model, "C3").value).toBe("C4");
+    expect(getEvaluatedCell(model, "C4").value).toBe("C6");
+  });
+
+  test("when formula, take into account the evaluated cell value", () => {
+    setSelection(model, ["A8:A9"]);
+    model.dispatch("REMOVE_DUPLICATES", { columnsToAnalyze: [0], hasHeader: false });
+
+    expect(getEvaluatedCell(model, "A8").value).toBe(4);
+    expect(getEvaluatedCell(model, "A9").value).toBe("");
+  });
+
+  test("when formula, update the references", () => {
+    setSelection(model, ["A2:A9"]);
+    model.dispatch("REMOVE_DUPLICATES", { columnsToAnalyze: [0], hasHeader: false });
+
+    expect(getEvaluatedCell(model, "A2").value).toBe(1);
+    expect(getEvaluatedCell(model, "A3").value).toBe(2);
+    expect(getEvaluatedCell(model, "A4").value).toBe(3);
+
+    expect(getEvaluatedCell(model, "A5").value).toBe(3);
+    expect(getCell(model, "A5")!.content).toBe("=A3+1");
+
+    expect(getEvaluatedCell(model, "A6").value).toBe("");
+    expect(getEvaluatedCell(model, "A7").value).toBe("");
+
+    expect(getEvaluatedCell(model, "A8").value).toBe("");
+    expect(getCell(model, "A8")).toBe(undefined);
+
+    expect(getEvaluatedCell(model, "A9").value).toBe("");
+  });
+
+  test("dont take into account the format", () => {
+    selectCell(model, "B2");
+    setFormat(model, "0.00%");
+
+    selectCell(model, "B4");
+    setFormat(model, "#,##0[$€]");
+
+    expect(getEvaluatedCell(model, "B2").value).toBe(42);
+    expect(getEvaluatedCell(model, "B2").format).toBe("0.00%");
+
+    expect(getEvaluatedCell(model, "B3").value).toBe(42);
+    expect(getEvaluatedCell(model, "B3").format).toBe(undefined);
+
+    expect(getEvaluatedCell(model, "B4").value).toBe(42);
+    expect(getEvaluatedCell(model, "B4").format).toBe("#,##0[$€]");
+
+    setSelection(model, ["B2:B4"]);
+    model.dispatch("REMOVE_DUPLICATES", { columnsToAnalyze: [1], hasHeader: false });
+
+    expect(getEvaluatedCell(model, "B2").value).toBe(42);
+    expect(getEvaluatedCell(model, "B2").format).toBe("0.00%");
+
+    expect(getEvaluatedCell(model, "B3").value).toBe("");
+    expect(getEvaluatedCell(model, "B3").format).toBe(undefined);
+
+    expect(getEvaluatedCell(model, "B4").value).toBe("");
+    expect(getEvaluatedCell(model, "B4").format).toBe(undefined);
+  });
+
+  test("can not remove duplicate with more than one selection", () => {
+    const cells = {
+      A1: { content: "24" },
+      A4: { content: "42" },
+      A6: { content: "242" },
+    };
+    model = new Model({ sheets: [{ id: "42", colNumber: 1, rowNumber: 5, cells: cells }] });
```

Comment:
Yeah if you have to initialize a new model with different cells, maybe it shouldn't be initialized in the beforeEach at all